allow compile time checking of string (missing placeholders, etc)

molar opened the pull request:
This change allows for compile time checking of the format string, such as an extra placeholder without an argument. tested on fmt 9.1.0

molar committed:
allow compile time checking of string (missing placeholders, etc)

xqms commented:
Interesting, thank you! The PR looks good to me. I'll test this over the weekend :)

xqms committed:
bump fmt to version 9.1.0

xqms committed:
keep rosfmt::format()

People might be using it.

xqms committed:
cmake: fix -> rosfmt9

xqms committed:
enable deprecated std::ostream operators

This is required e.g. for Eigen (see #2). If we don't do this, existing
packages may break.

xqms commented:
I've had to:
 * actually bump the fmt version to 9.1.0
 * enable `FMT_DEPRECATED_OSTREAM` so that people using this functionality are not surprised
 * keep the `rosfmt::format()` function around in case anyone is using it

Now waiting for #8 to complete so that we have a proper CI in case before merging.

xqms commented:
Thank you for contributing!
